cylc/flow/task_job_mgr.py

```diff
@@ -238,7 +238,6 @@ def prep_submit_task_jobs(self, workflow, itasks, check_syntax=True):
                 itask.submit_num += 1
                 itask.state_reset(TASK_STATUS_PREPARING)
                 self.data_store_mgr.delta_task_state(itask)
-                self.workflow_db_mgr.put_update_task_state(itask)
```

Comment:
This means the DB won't pick up on the PREPARING state until later on?

I think a `state_reset` should always be followed by a DB update?

Reply:
OK, I've gone back for a look at this.  The reason is, as per the comment added to the removed method:
```python
        This method is only needed as a final update for finished tasks,
        when they get removed from the task_pool.
```
The task pool and task states table is currently updated at the same time as the datastore, in the main loop via `Scheduler.update_data_structure()`

We have evidently been calling `put_update_task_state` in a couple of places in addition to that, but given the above that wasn't the intention.

We've been planning to all of these updates in more event-driven way, but that's beyond the scope of this PR and will required some discussion.  (Maybe should post a new issue for thoughts on this?).

Reply:
Ok, that makes sense.